tests/test_serializers.py

```diff
@@ -255,6 +231,204 @@ def test_view_project_pending_registration_for_write_contributor_does_not_contai
         pending_reg.remove()
 
 
+class TestViewProjectEmbeds(OsfTestCase):
+
+    def setUp(self):
+        super(TestViewProjectEmbeds, self).setUp()
+        self.user = UserFactory()
+        self.project = ProjectFactory(creator=self.user)
+
+    def test_view_project_embed_forks_excludes_registrations(self):
+        project = ProjectFactory()
+        fork = project.fork_node(Auth(project.creator))
+        reg = RegistrationFactory(project=fork)
+
+        res = _view_project(project, auth=Auth(project.creator), embed_forks=True)
+
+        assert_in('forks', res['node'])
+        assert_equal(len(res['node']['forks']), 1)
+
+        assert_equal(res['node']['forks'][0]['id'], fork._id)
+
+    # Regression test for https://github.com/CenterForOpenScience/osf.io/issues/1478
+    @mock.patch('website.archiver.tasks.archive')
+    def test_view_project_embed_registrations_includes_contribution_count(self, mock_archive):
+        self.project.register_node(get_default_metaschema(), Auth(user=self.project.creator), '', None)
+        data = _view_project(node=self.project, auth=Auth(self.project.creator), embed_registrations=True)
+        assert_is_not_none(data['node']['registrations'][0]['nlogs'])
+
+    # Regression test
+    def test_view_project_embed_registrations_sorted_by_registered_date_descending(self):
+        # register a project several times, with various registered_dates
+        registrations = []
+        for days_ago in (21, 3, 2, 8, 13, 5, 1):
+            registration = RegistrationFactory(project=self.project)
+            reg_date = registration.registered_date - dt.timedelta(days_ago)
+            registration.registered_date = reg_date
+            registration.save()
+            registrations.append(registration)
+
+        registrations.sort(key=lambda r: r.registered_date, reverse=True)
+        expected = [r._id for r in registrations]
+
+        data = _view_project(node=self.project, auth=Auth(self.project.creator), embed_registrations=True)
+        actual = [n['id'] for n in data['node']['registrations']]
+        assert_equal(actual, expected)
+
+    def test_view_project_embed_descendants(self):
+        child = NodeFactory(parent=self.project, creator=self.user)
+        res = _view_project(self.project, auth=Auth(self.project.creator), embed_descendants=True)
+        assert_in('descendants', res['node'])
+        assert_equal(len(res['node']['descendants']), 1)
+        assert_equal(res['node']['descendants'][0]['id'], child._id)
+
+
+class TestGetReadableDescendants(OsfTestCase):
```

Comment:
These used to be view tests. I removed `/api/v1/<nid>/get_readable_descendants/`, so I ported the tests into serializer tests.